.github/workflows/tests.yml

```diff
@@ -0,0 +1,71 @@
+name: Tests
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  tests:
+    name: ${{ matrix.os }}, Python ${{ matrix.python-version }}${{ matrix.experimental && ', experimental' || '' }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.8", "3.9", "3.10"]
+        experimental: [false]
+        include:
+          - os: "ubuntu-latest"
+            python-version: "3.10"
+            experimental: true
+
+          - os: "ubuntu-latest"
+            python-version: "3.10"
+            experimental: false
+            coverage: true
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if echo ${{ matrix.experimental }} | grep -c "true"
+          then
+            pip install -e .[dev]
+          else
+            pip install -r requirements.txt
+            pip install -e .
+          fi
+        shell: bash
+
+      - name: Test with pytest

```

Comment:
This also run the unit tests or just check the coverage?

Reply:
Tests are also runned and failure is dectected as usual